### PR TITLE
Remove livecheck blocks for deprecated formulae

### PR DESCRIPTION
**Pull request by samford**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This removes the `livecheck` blocks from deprecated formulae that aren't going to be updated in the future. `livecheck` blocks should be kept (or added) for any deprecated formulae that will continue to receive updates but that's not the case for any of these.

Livecheck will skip deprecated formulae unless the formula contains a `livecheck` block. In this case, we don't want to check these formulae, so it's appropriate to remove the `livecheck` blocks so they're skipped.

I suggested that we may want to create a RuboCop for this (along with an allowlist, to allow some formulae to be updated despite deprecation) in https://github.com/Homebrew/brew/pull/8842#issuecomment-703105746.

---

Besides that, this also addresses the following issues:

* Update the license for `archey` from `GPL-2.0` to `GPL-2.0-or-later` (referencing https://github.com/obihann/archey-osx/blob/master/LICENSE)
* Remove `erlang@20` resource mirrors where the files are no longer found

### CODE REVIEW

**Comment by zachauten** [comment]
Why would a formula that's still being updated be marked deprecated? I'm curious about how many formula there are where this is the case.

**Review by zachauten — APPROVED** [review]

**Comment by SMillerDev** [comment]
They could be deprecated because of outdated dependencies.

**Comment by samford** [comment]
> Why would a formula that's still being updated be marked deprecated? I'm curious about how many formula there are where this is the case.

@zachauten I went through all the formulae that currently contain `deprecate!` and the only two I found that were updated after being deprecated were `helm@2` and `libpqxx@6`. Neither of these have specific deprecation dates, which is part of the issue.

The `helm@2` formula was deprecated with `deprecate! because: :unsupported` on 2020-09-16 since [v2 isn't being developed further](https://github.com/helm/helm/issues/2466#issuecomment-691177445) (only patched). However, `helm` v2 isn't officially deprecated yet and there have been two patch updates since then. There's a [first-party blog post explaining the timeline for the `helm` v2 deprecation](https://helm.sh/blog/helm-v2-deprecation-timeline/), including the date the deprecation will take effect (2020-11-13). I created a PR (#62200) to add the deprecation date and restore the `livecheck` block until the deprecation takes effect (along with a comment that the `livecheck` block should be removed after that point).

The `libpqxx@6` formula was deprecated with `deprecate! because: :versioned_formula` when it was created on 2020-05-20. It has been updated once, on 2020-06-24. There's a chance it may see additional updates in the future if they continue to release v6 patches but I couldn't find a clear deprecation date from a quick search. If anyone finds the date, it would be good to add it.

Long story short, it's not very common that a deprecated formula is updated but it does happen for some of the open-ended instances of `deprecate!`.
